airbyte-integrations/bases/base-typing-deduping-test/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.java

```diff
@@ -363,11 +364,46 @@ public void testLineBreakCharacters() throws Exception {
         .build());
   }
 
+  /**
+   * Run a sync, then remove the {@code name} column from the schema and run a second sync. Verify
+   * that the final table doesn't contain the `name` column after the second sync.
+   */
   @Test
-  @Disabled("Not yet implemented")
   public void testIncrementalSyncDropOneColumn() throws Exception {
-    // TODO in incremental dedup mode: run a sync, remove a column from the schema, run another sync
-    // verify that the column is dropped from the destination table
+    AirbyteStream stream = new AirbyteStream()
+        .withNamespace(streamNamespace)
+        .withName(streamName)
+        .withJsonSchema(SCHEMA);
+    ConfiguredAirbyteCatalog catalog = new ConfiguredAirbyteCatalog().withStreams(List.of(
+        new ConfiguredAirbyteStream()
+            .withSyncMode(SyncMode.INCREMENTAL)
+            .withCursorField(List.of("updated_at"))
+            .withDestinationSyncMode(DestinationSyncMode.APPEND)
+            .withStream(stream)));
+
+    // First sync
+    List<AirbyteMessage> messages1 = readMessages("sync1_messages.jsonl");
+
+    runSync(catalog, messages1);
+
+    List<JsonNode> expectedRawRecords1 = readRecords("sync1_expectedrecords_nondedup_raw.jsonl");
+    List<JsonNode> expectedFinalRecords1 = readRecords("sync1_expectedrecords_nondedup_final.jsonl");
+    verifySyncResult(expectedRawRecords1, expectedFinalRecords1);
+
+    // Second sync
+    List<AirbyteMessage> messages2 = readMessages("sync2_messages.jsonl");
+    JsonNode trimmedSchema = SCHEMA.deepCopy();
+    ((ObjectNode) trimmedSchema.get("properties")).remove("name");
+    stream.setJsonSchema(trimmedSchema);
+
+    runSync(catalog, messages2);
+
+    // The raw data is unaffected by the schema, but the final table should not have a `name` column.
+    List<JsonNode> expectedRawRecords2 = readRecords("sync2_expectedrecords_fullrefresh_append_raw.jsonl");
+    List<JsonNode> expectedFinalRecords2 = readRecords("sync2_expectedrecords_fullrefresh_append_final.jsonl").stream()
+        .peek(record -> ((ObjectNode) record).remove("name"))
```

Comment:
I could be convinced to put this in its own file, not sure if we'd prefer being more explicit vs more duplicates of these record files